client/src/components/Header.js

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from "react";
+import { NavLink } from "react-router-dom";
+import { connect } from "react-redux";
+import { logout } from "../store/actions/userActions";
+
+const iStyle = {
+  fontSize: "30px"
+};
+
+class Header extends Component {
+  constructor() {
+    super();
+    this.state = {
+      isOpen: false
+    };
+    this.logout = this.logout.bind(this);
+  }
+
+  toggleModal = () => {
+    if (!this.state.isOpen) {
+      document.addEventListener("click", this.toggleModal, false);
```

Comment:
You should try to keep listeners in the React api.  It might get confusing since this is bound to the document and will capture every click.